src/frontends/tensorflow_common/src/op/scatter_nd.cpp

```diff
@@ -15,15 +15,28 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_scatter_nd_op(const NodeContext& node) {
-    default_op_checks(node, 3, {"ScatterNd", "SCATTER_ND"});
+    default_op_checks(node, 3, {"ScatterNd", "SCATTER_ND"}, true);
     auto input_indices = node.get_input(0);
     auto updates = node.get_input(1);
     auto shape = node.get_input(2);
 
+    auto complex_type_mark_updates = as_type_ptr<ComplexTypeMark>(updates.get_node_shared_ptr());
+
+    if (complex_type_mark_updates) {
+        updates = complex_type_mark_updates->input_value(0);
+    }
+
     auto input_data = create_same_type_const<int32_t>(updates, vector<int32_t>{0}, Shape{1});
     auto broadcast = make_shared<v3::Broadcast>(input_data, shape);
```

Comment:
`broadcast` should also have `2` auxiliary dimension so `shape` needs to be updated.
Please implement complex type and regular type separately, i.e. if (complex_type_mark_updates) {impl...} else {existing logic}